src/routes/accounts.py

```
@@ -1,25 +1,237 @@
+async def reset_password_request(
+    data: PasswordResetCompleteRequestSchema,
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Request a password reset token.
+    Returns generic message for security (prevents email enumeration).
+    Token is only created for active users.
+    """
+    try:
+        result = await db.execute(
+            select(UserModel)
+            .options(joinedload(UserModel.password_reset_token))
+            .where(UserModel.email == data.email)
+        )
+        user = result.unique().scalar_one_or_none()
+
+        message = "If you are registered, you will receive an email with instructions."
+
+        # Токен создаётся только для существующих И активных пользователей
+        if not user or not user.is_active:
+            return {"message": message}
+
+        # Удаляем старый токен
+        if user.password_reset_token:
+            await db.delete(user.password_reset_token)
+            await db.flush()
+
+        # Создаём новый токен
+        reset_token = PasswordResetTokenModel(user_id=user.id)
+        db.add(reset_token)
+        await db.commit()
+
+        return {"message": message}
+
+    except SQLAlchemyError:
+        raise HTTPException(
+            status_code=500,
+            detail="An error occurred while processing the request."
+        )
+
+@router.post("/reset-password/complete/", response_model=MessageResponseSchema)
+async def reset_password_complete(
+    data: PasswordResetRequestSchema,
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Complete password reset using token.
+    Validates token, updates password, and removes the token.
+    """
+    try:
+        result = await db.execute(
+            select(UserModel)
+            .options(joinedload(UserModel.password_reset_token))
+            .where(UserModel.email == data.email)
+        )
+        user = result.unique().scalar_one_or_none()
+
+        if not user or not user.password_reset_token:
```

The requirements state that you should return an error if the user is inactive. The current logic only checks for the user's existence. Please add a check for `user.is_active` here to fully meet the requirements.